lms/views/api/canvas/authorize.py

```diff
@@ -127,15 +128,24 @@ def oauth2_redirect(request):
     request_method="GET",
     route_name="canvas_api.oauth.authorize",
     renderer="lms:templates/api/oauth2/redirect_error.html.jinja2",
+    tm_active=True,
 )
 def oauth2_redirect_error(request):
     kwargs = {
         "auth_route": "canvas_api.oauth.authorize",
         "canvas_scopes": list(ALL_SCOPES),
     }
-
     if request.params.get("error") == "invalid_scope":
-        kwargs["error_code"] = request.context.js_config.ErrorCode.CANVAS_INVALID_SCOPE
+        error_code = request.context.js_config.ErrorCode.CANVAS_INVALID_SCOPE
+        kwargs["error_code"] = error_code
+
+        EventService.queue_event(
+            LTIEvent(
+                request=request,
+                type=LTIEvent.Type.ERROR_CODE,
+                data={"code": error_code},
```

Comment:
Do we get any additional info back about _which_ scopes were missing?

Reply:
No, not currently because we are not parsing the error message from canvas.

If we ever go that route we could easily dump that info into the event and, more importantly, improve the error message displayed to the user.